p03-form-validations/src/Form/Form.component.js

```diff
@@ -1,23 +1,62 @@
-import React from 'react'
-import { Card } from '../Shared/components/Card'
-import { Button, Input } from '../Shared/components'
-import { fieldsList } from './Form.services'
+import React, { useState, useEffect } from "react";
+import { Card } from "../Shared/components/Card";
+import { Button, Input } from "../Shared/components";
+import { fieldsList, isFieldValid, handleSubmitForm } from "./Form.services";
 
-import "./Form.styles.css"
+import "./Form.styles.css";
+const initialState = fieldsList.map((field) => ({
+  ...field,
+  isValid: false,
+  value: "",
+}));
 
 const Form = () => {
-  
+  const [state, setState] = useState(initialState);
+  const [disableButton, setdisableButton] = useState(true);
+
+  useEffect(() => {
+    setdisableButton(state.some((field) => !field.isValid));
+    console.log(state.some((field) => !field.isValid));
```

Comment:
Logs are very helpful debugging and solving problems, but we should avoid pushing them to repositories, because it may go to production code